diff --git a/Content.Shared/_Moffstation/Cards/Systems/SharedPlayingCardsSystem.Card.cs b/Content.Shared/_Moffstation/Cards/Systems/SharedPlayingCardsSystem.Card.cs
index b501e60431..b1fe0fe2fb 100644
--- a/Content.Shared/_Moffstation/Cards/Systems/SharedPlayingCardsSystem.Card.cs
+++ b/Content.Shared/_Moffstation/Cards/Systems/SharedPlayingCardsSystem.Card.cs
@@ -256,9 +256,9 @@ private static bool SetOrInvert(ref bool value, bool? setToValue)
     private bool FlipCardInDeck(PlayingCardInDeck card, bool? faceDown = null) => card switch
     {
         PlayingCardInDeckNetEnt(var cardNetEnt) =>
-            NetEntToCard(cardNetEnt) is { } cardEnt && SetFacingOrFlip(cardEnt, null),
+            NetEntToCard(cardNetEnt) is { } cardEnt && SetFacingOrFlip(cardEnt, faceDown),
         PlayingCardInDeckUnspawnedData(var data, _, _) => SetOrInvert(ref data.FaceDown, faceDown),
-        PlayingCardInDeckUnspawnedRef(_, var fd) => SetOrInvert(ref fd, faceDown),
+        PlayingCardInDeckUnspawnedRef unspawnedRef => SetOrInvert(ref unspawnedRef.FaceDown, faceDown),
         _ => card.ThrowUnknownInheritor<PlayingCardInDeck, bool>(),
     };
 
