diff --git a/Content.Shared/_Moffstation/Clothing/ModularHud/Systems/SharedModularHudSystem.cs b/Content.Shared/_Moffstation/Clothing/ModularHud/Systems/SharedModularHudSystem.cs
index 7e2eb5f124..2c06589543 100644
--- a/Content.Shared/_Moffstation/Clothing/ModularHud/Systems/SharedModularHudSystem.cs
+++ b/Content.Shared/_Moffstation/Clothing/ModularHud/Systems/SharedModularHudSystem.cs
@@ -208,12 +208,13 @@ private void OnInteractUsing(Entity<ModularHudComponent> entity, ref InteractUsi
 
             if (!usedHasQuality)
             {
-                Loc.GetString(
-                    usedHasQuality
-                        ? entity.Comp.RemoveModulesVerbMessage
-                        : entity.Comp.MissingToolQualityErrorText,
-                    ("quality", Loc.GetString(toolQuality?.Name ?? "Unknown")),
-                    ("hud", Name(entity))
+                _popup.PopupPredictedCursor(
+                    Loc.GetString(
+                        entity.Comp.MissingToolQualityErrorText,
+                        ("quality", Loc.GetString(toolQuality?.Name ?? "Unknown")),
+                        ("hud", Name(entity))
+                    ),
+                    args.User
                 );
                 return;
             }
@@ -265,6 +266,9 @@ private void OnGetInteractionVerbs(Entity<ModularHudComponent> entity, ref GetVe
                 Text = Loc.GetString(entity.Comp.InsertModuleVerbText),
                 Act = () =>
                 {
+                    if (disabledReason != null)
+                        return;
+
                     if (!_container.Insert(used, entity.Comp.ModuleContainer))
                     {
                         // This should always succeed, so error out if it doesn't.
